lib/driver_sht41/sht41.c

```diff
@@ -131,43 +125,69 @@ static void sht41_on_read(const struct async_i2c_request *req) {
     local->reading_expiration = make_timeout_time_ms(SHT41_INVALID_TIMEOUT_MS);
 }
 
-static void sht41_on_error(const sht41_error_code error_type, uint32_t i2c_error_code) {
+static void sht41_on_error(const sht41_error_code error_type) {
     err_count++;
-    if (err_count >= 3) {
+    if (err_count >= ACCEPTABLE_ERR_COUNT) {
         if (error_cb) {
-            error_cb(error_type, i2c_error_code);
+            error_cb(error_type);
         }
     }
 }
 
 // called by the i2c
-static void on_async_i2c_error(__unused struct async_i2c_request *req, uint32_t error_code) {
-    on_error(SHT41_ERROR_I2C_COMPLAINT, error_code);
+static void on_async_i2c_error(__unused const struct async_i2c_request *req, __unused uint32_t error_code) {
+    sht41_on_error(SHT41_ERROR_I2C_COMPLAINT);
 }
 
-void sht41_init(sht41_error_cb board_error_cb) {
+void sht41_init(sht41_error_cb board_error_cb, uint8_t board, uint8_t addr) {
```

Comment:
According to the datasheet, the only address this device can have is 0x44. If this is the case, then no need to accept addr as an argument, as it'll be constant.